sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/AadAuthorizationTests.java

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.rx;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.cosmos.CosmosAsyncClient;
+import com.azure.cosmos.CosmosAsyncContainer;
+import com.azure.cosmos.CosmosAsyncDatabase;
+import com.azure.cosmos.CosmosClientBuilder;
+import com.azure.cosmos.CosmosDatabaseForTest;
+import com.azure.cosmos.implementation.InternalObjectNode;
+import com.azure.cosmos.implementation.TestConfigurations;
+import com.azure.cosmos.implementation.Utils;
+import com.azure.cosmos.models.CosmosContainerResponse;
+import com.azure.cosmos.models.CosmosDatabaseResponse;
+import com.azure.cosmos.models.CosmosItemRequestOptions;
+import com.azure.cosmos.models.CosmosItemResponse;
+import com.azure.cosmos.models.CosmosQueryRequestOptions;
+import com.azure.cosmos.models.PartitionKey;
+import com.azure.cosmos.util.CosmosPagedFlux;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.time.OffsetDateTime;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Properties;
+import java.util.UUID;
+
+public class AadAuthorizationTests extends TestSuiteBase {
+    private final static Logger log = LoggerFactory.getLogger(AadAuthorizationTests.class);
+    private static final ObjectMapper OBJECT_MAPPER = Utils.getSimpleObjectMapper();
+
+    private final static String PARTITION_KEY_PATH = "/mypk";
+    private final String databaseId = CosmosDatabaseForTest.generateId();
+
+    protected AadAuthorizationTests() {
+    }
+
+    @Test(groups = { "emulator" }, timeOut = 10 * TIMEOUT)
+    public void createAadTokenCredential() throws InterruptedException {
+        CosmosAsyncDatabase db = null;
+
+        CosmosAsyncClient cosmosAsyncClient = new CosmosClientBuilder()
+            .endpoint(TestConfigurations.COSMOS_EMULATOR_HOST)
```

Comment:
why are we defining new public property for something which is already available?

I thought the plan was to reuse `TestConfiguration.HOST` and not redefine a constant for the same thing.

1) on the CI the value for `TestConfiguration.HOST` will be populated by the CI.
2) in local debugging that can be populated from a properties file from user home.

This breaks both 1) and 2).

Reply:
This is an emulator only test which it cannot be run against prod endpoint.